Extract TensorPipeAgent's collectNames to be a standalone utility function

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #53209 Use Store collect and verify names in all RPC agents
* **#53202 Extract TensorPipeAgent's collectNames to be a standalone utility function**

Differential Revision: [D26791525](https://our.internmc.facebook.com/intern/diff/D26791525)

mrshenli committed:
Extract TensorPipeAgent's collectNames to be a standalone utility function

[ghstack-poisoned]

mrshenli committed:
Update on "Extract TensorPipeAgent's collectNames to be a standalone utility function"




[ghstack-poisoned]

H-Huang commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -770,7 +770,6 @@ def test_register_rpc_backend_and_set_and_start_rpc_backend(
             rpc_backend_options=self.rpc_backend_options,
         )
 
-    @dist_init(setup_rpc=False)
```
Was this intentional?

mrshenli replied:
this is not intentional, my bad. Thanks for spotting this.

H-Huang commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -787,6 +786,17 @@ def test_duplicate_name(self):
+    def test_duplicate_name_2(self):
+        with self.assertRaisesRegex(RuntimeError, "is not unique"):
+            rpc.init_rpc(
+                name=worker_name(self.rank % 2),
```
nit: Would an error not be raised if world_size <= 2? Maybe it can be `self.rank % (self.world_size - 1)` if that is the case

rohan-varma commented on torch/csrc/distributed/rpc/agent_utils.h:
```diff
@@ -0,0 +1,49 @@
+        nameToId.find(workerName) == nameToId.end(),
+        "RPC worker name ",
+        workerName,
+        " is not unique.");
```
Nit: to help with debugging, it might be useful to log the ids for the duplicate names here: `nameToId.find(workerName)->second` and `workerId`.

mrshenli replied:
good point, will add.

mrshenli replied:
will add this to the top PR to avoid modifying this twice during rebase.

mrshenli committed:
Update on "Extract TensorPipeAgent's collectNames to be a standalone utility function"


Differential Revision: [D26791525](https://our.internmc.facebook.com/intern/diff/D26791525)

[ghstack-poisoned]

mrshenli committed:
Update on "Extract TensorPipeAgent's collectNames to be a standalone utility function"


Differential Revision: [D26791525](https://our.internmc.facebook.com/intern/diff/D26791525)

[ghstack-poisoned]
